spec/models/claim_spec.rb

```diff
@@ -29,6 +32,13 @@
     it { is_expected.to have_many(:mentors).through(:mentor_trainings) }
   end
 
+  context "with validations" do
+    it "validates uniqueness of reference" do
+      claim = build(:claim)
+      expect(claim).to validate_uniqueness_of(:reference).allow_nil
+    end
```

Comment:
You can use an implicit subject here.

```suggestion
    it { is_expected.to validate_uniqueness_of(:reference).allow_nil }
```

Reply:
I need to build the claim so that it links to a school. The claim cannot exist without a link to a school. I get a PG error saying school_id needs to exist as the validate uniqueness saves the record to db to check if the validation is actually working.

Reply:
If you override `subject`, `is_expected` will automatically reference it too.